tests/admin_views/tests.py

```diff
@@ -3687,21 +3687,17 @@ class AdminInlineFileUploadTest(TestCase):
     @classmethod
     def setUpTestData(cls):
         cls.superuser = User.objects.create_superuser(username='super', password='secret', email='super@example.com')
-
-    def setUp(self):
-        self.client.force_login(self.superuser)
-
-        # Set up test Picture and Gallery.
-        # These must be set up here instead of in fixtures in order to allow Picture
-        # to use a NamedTemporaryFile.
         file1 = tempfile.NamedTemporaryFile(suffix=".file1")
         file1.write(b'a' * (2 ** 21))
         filename = file1.name
         file1.close()
-        self.gallery = Gallery(name="Test Gallery")
-        self.gallery.save()
-        self.picture = Picture(name="Test Picture", image=filename, gallery=self.gallery)
-        self.picture.save()
+        cls.gallery = Gallery(name="Test Gallery")
+        cls.gallery.save()
+        cls.picture = Picture(name="Test Picture", image=filename, gallery=cls.gallery)
+        cls.picture.save()
```

Comment:
Maybe `.objects.create(...)`? We will have two lines less :smile: Please change to single quotes.